### PR TITLE
feat(keybindings): add key notation parser (String <-> KeyChord)

### LINKED ISSUE #96: [Feature] Key Notation Parser (String ↔ KeyChord)
## Summary

Build a key notation parser that converts human-readable strings like `"C-n"`, `"S-Tab"`, `"M-x"` into `KeyChord` structs. This is the foundational building block for configurable keybindings.

**Parent**: [Epic] Configurable Keybindings #89

## Problem / Background

The existing `KeybindingConfig` (`config.rs:234-277`) stores keybindings as strings like `"C-h ?"` and `"SPC"`, but there is NO parsing layer to convert these into the `KeyChord` struct used by `KeyHandler`.

Current `KeyChord` definition (`keybindings.rs:12-38`):
```rust
pub struct KeyChord {
    pub key: KeyCode,
    pub modifiers: KeyModifiers,
}
```

Constructor helpers exist (`none()`, `ctrl()`, `shift()`, `alt()`) but no string→KeyChord parser.

### Key Notation Format (following Helix/Zellij conventions)

| Notation | Meaning | Example |
|----------|---------|---------|
| `C-x` | Ctrl + x | `C-n` = Ctrl+N |
| `S-x` | Shift + x | `S-Tab` = Shift+Tab |
| `A-x` or `M-x` | Alt + x | `M-x` = Alt+X |
| `C-S-x` | Ctrl + Shift + x | `C-S-d` |
| `F1`-`F12` | Function keys | `F1` |
| `Space` or `SPC` | Space key | `Space` |
| `Enter`, `Tab`, `Esc`, `Backspace` | Named keys | `Enter` |
| `Up`, `Down`, `Left`, `Right` | Arrow keys | `Up` |
| `Home`, `End`, `PgUp`, `PgDn` | Navigation keys | `PgDn` |
| Single char | Literal key | `q`, `j`, `?` |

### Implementation

Create `src/ui/key_parser.rs`:

```rust
use crossterm::event::{KeyCode, KeyModifiers};
use crate::ui::keybindings::KeyChord;

#[derive(Debug, thiserror::Error)]
pub enum KeyParseError {
    #[error("Unknown key name: {0}")]
    UnknownKey(String),
    #[error("Empty key notation")]
    Empty,
    #[error("Invalid modifier: {0}")]
    InvalidModifier(String),
}

pub fn parse_key_notation(notation: &str) -> Result<KeyChord, KeyParseError> {
    let notation = notation.trim();
    if notation.is_empty() {
        return Err(KeyParseError::Empty);
    }

    let mut modifiers = KeyModifiers::empty();
    let parts: Vec<&str> = notation.split('-').collect();

    // Parse modifiers (C-, S-, A-/M- prefixes)
    let key_part = parse_modifiers(&parts, &mut modifiers)?;

    // Parse the key itself
    let key = parse_key_code(key_part)?;

    Ok(KeyChord { key, modifiers })
}

/// Serialize KeyChord back to notation string
pub fn key_to_notation(chord: &KeyChord) -> String {
    // Inverse of parse_key_notation
}
```

### Also implement the inverse
`key_to_notation(chord: &KeyChord) -> String` for displaying keybindings in help text and error messages.

## Files to Modify

| File | Change |
|------|--------|
| `src/ui/key_parser.rs` | **NEW FILE**: Key notation parser + serializer |
| `src/ui/mod.rs` | Add `pub mod key_parser;` |

## Acceptance Criteria

- [ ] `parse_key_notation("C-n")` → `KeyChord { key: Char('n'), modifiers: CONTROL }`
- [ ] `parse_key_notation("S-Tab")` → `KeyChord { key: Tab, modifiers: SHIFT }`
- [ ] `parse_key_notation("F1")` → `KeyChord { key: F(1), modifiers: NONE }`
- [ ] `parse_key_notation("q")` → `KeyChord { key: Char('q'), modifiers: NONE }`
- [ ] `parse_key_notation("C-S-d")` → `KeyChord { key: Char('d'), modifiers: CONTROL | SHIFT }`
- [ ] `key_to_notation(chord)` round-trips correctly for all supported notations
- [ ] Invalid notations return descriptive `KeyParseError`
- [ ] `cargo test && cargo clippy -- -D warnings` passes

### Unit Tests (comprehensive — this is a parser)
```rust
#[test]
fn test_parse_single_char() {
    let chord = parse_key_notation("q").unwrap();
    assert_eq!(chord.key, KeyCode::Char('q'));
    assert_eq!(chord.modifiers, KeyModifiers::NONE);
}

#[test]
fn test_parse_ctrl_modifier() {
    let chord = parse_key_notation("C-n").unwrap();
    assert_eq!(chord.key, KeyCode::Char('n'));
    assert_eq!(chord.modifiers, KeyModifiers::CONTROL);
}

#[test]
fn test_parse_function_key() {
    let chord = parse_key_notation("F3").unwrap();
    assert_eq!(chord.key, KeyCode::F(3));
}

#[test]
fn test_parse_combined_modifiers() {
    let chord = parse_key_notation("C-S-d").unwrap();
    assert!(chord.modifiers.contains(KeyModifiers::CONTROL));
    assert!(chord.modifiers.contains(KeyModifiers::SHIFT));
}

#[test]
fn test_roundtrip() {
    let original = "C-S-Tab";
    let chord = parse_key_notation(original).unwrap();
    assert_eq!(key_to_notation(&chord), original);
}

#[test]
fn test_error_empty() {
    assert!(parse_key_notation("").is_err());
}

#[test]
fn test_error_unknown() {
    assert!(parse_key_notation("C-FOOBAR").is_err());
}
```

## Definition of Done

- [ ] Key notation parser handles all documented notations
- [ ] Inverse serializer produces canonical notation
- [ ] 15+ unit tests covering all key types, modifiers, edge cases, and errors
- [ ] Code passes linting (clippy) with no warnings
- [ ] Code formatted with rustfmt

### PR DESCRIPTION
**Pull request by lqdev**
## Summary

Closes #96 — adds the key notation parser that converts human-readable strings like `C-n`, `S-Tab`, `M-x` into `KeyChord` structs and back. This is the foundational building block for configurable keybindings (epic #89).

## Changes

- `src/ui/key_parser.rs` (**new**): `parse_key_notation()`, `key_to_notation()`, `KeyParseError` type
- `src/ui/mod.rs`: added `pub mod key_parser;`
- `CHANGELOG.md`: new `[Unreleased] > Added` entry
- `.github/skills/work-on-issue/SKILL.md`: added `## Manual Testing` section requirement to PR template

## Notation Supported

| Notation | Meaning |
|----------|---------|
| `C-x` | Ctrl + x |
| `S-x` | Shift + x |
| `A-x` / `M-x` | Alt + x |
| `C-S-x` | Ctrl + Shift + x |
| `F1`–`F12` | Function keys |
| `Space` / `SPC` | Space bar |
| `Enter`, `Tab`, `Esc`, `Backspace`, `Delete` | Named keys |
| `Up`, `Down`, `Left`, `Right` | Arrow keys |
| `Home`, `End`, `PgUp`, `PgDn` | Navigation keys |
| Single char | Literal (`q`, `?`, `3`) |

## Manual Testing

N/A — key notation parser is not wired to the keybinding system yet. It is a pure parsing module with no UI surface. Consumers will be added in #97 (Expand KeybindingConfig) and #98 (Load Keybindings from Config at Runtime).

## Tests

31 unit tests inside `key_parser.rs` covering:
- All key types (chars, named keys, function keys, arrow/nav keys)
- All modifier prefixes (C-, S-, A-, M-, combinations)
- Round-trip: `key_to_notation(parse_key_notation(s)) == s`
- Error cases: empty input, whitespace-only, unknown key name, F0/F13 out of range
- Edge cases: bare `-`, leading/trailing whitespace

## Quality

- `cargo fmt` ✅
- `cargo clippy -- -D warnings` ✅
- `cargo test` ✅ (301 unit tests pass; 1 pre-existing unrelated failure in `test_opml_local_file` — missing local file, not introduced by this PR)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,7 +17,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- **Foundation for configurable keybindings**: new `key_parser` module (`src/ui/key_parser.rs`) converts human-readable key notation strings to `KeyChord` structs and back
+  - Supports modifier prefixes: `C-` (Ctrl), `S-` (Shift), `A-`/`M-` (Alt), and combinations like `C-S-x`
+  - Named keys: `Enter`, `Tab`, `Esc`, `Backspace`, `Delete`, `Space`/`SPC`, arrow keys, `Home`, `End`, `PgUp`/`PgDn`
+  - Function keys `F1`–`F12` with optional modifier prefix (e.g., `C-F3`)
+  - Single character literals (e.g., `q`, `?`, `3`)
+  - `key_to_notation()` serializes a `KeyChord` back to its canonical string (full round-trip support)
+  - Descriptive `KeyParseError` variants for empty input, unknown key names, and missing keys after modifier
+  - Tests added: 31 unit tests covering all key types, modifiers, combinations, round-trips, and error cases. Closes [#96](https://github.com/lqdev/podcast-tui/issues/96).
+
+- Added vim-style navigation keys`j`/`k` (down/up), `g` (top), `G` (bottom) as aliases for arrow keys, Home, and End in all list buffers. Also added `C-n`/`C-p` as global Emacs-style navigation aliases. Closes [#93](https://github.com/lqdev/podcast-tui/issues/93).
```
[the remark this answers is not in the document]
Comment (lqdev) [reply CHANGELOG.md]: Accepted — fixed in commit 3cc3560. The space between "keys" and the backtick was indeed missing in the raw file (keys`j`/`k` with no space). Pre-existing typo from the #93 CHANGELOG entry; cleaned up while we were already touching CHANGELOG.md for this PR.
